.github/workflows/ci.yml

```diff
@@ -79,4 +96,28 @@ jobs:
           make -j4
           make install
           cd ../../../
+      - name: Build Software
+        run: |
           make -C target/snitch_cluster SELECT_RUNTIME=banshee sw
+      - name: Run Unit Tests
+        working-directory: target/snitch_cluster
+        env:
+          SNITCH_LOG: info
+        run: |-
+          ./sw/tests/run.py sw/tests/banshee-apps.list --simulator banshee \
+          2>&1 | tee tests.log
```

Comment:
Why do we need to tee the output to a file and grep the file to verify that the tests passed?

Reply:
Because the python return value was not forwarded to the CI runner. E.g. also on Gitlab VCS never really run and all tests failed, but it was reported as successful. This makes it a very explicit check that cannot be broken without being noticed.

Reply:
You mean the Python script was not returning an non-zero exit code upon VCS failing? I think we should fix that instead of adding more logic